projects/gradio_demo/launch.py

```diff
@@ -58,6 +59,11 @@ def get_instance(cls, instance_name, callback: Callable):
         if len(cls._cache) == cls.max_size:
             cls._cache.pop(cls.max_size - 1)
             torch.cuda.empty_cache()
+            try:
+                import torch.mps
+                torch.mps.empty_cache()
+            except ImportError:
+                pass
```

Comment:
Here, I empty the cuda cache, so that `torch.cuda.mem_get_info` can get the true available GPU memory, which can help to select the GPU to use.
But for MPS devices, do we need to empty the cache?

Reply:
Hi you are right, we do not need to clear mps cache.